### PR TITLE
tests: add iter-remote-refs test

### PR DESCRIPTION
**Pull request by dtrifiro**
- add test for `iter-remote-refs` refs (broken for `dulwich<0.20.38`)
- bump dulwich version to `>=0.20.38`

Related https://github.com/iterative/dvc/issues/7670

### CODE REVIEW

File: tests/test_git.py [hunk 1]
```diff
@@ -417,6 +422,49 @@ def test_fetch_refspecs(
     assert baz_rev == scm.get_ref("refs/foo/baz")
 
 
+@pytest.mark.skip_git_backend("pygit2", "gitpython")
+@pytest.mark.parametrize("use_url", [True, False])
+@pytest.mark.xfail(
+    os.name == "nt",
+    reason="file:// urls are not supported by dulwich on windows",
```
Comment (dtrifiro) [inline tests/test_git.py:429]: This xfail can be removed when https://github.com/jelmer/dulwich/pull/965 is merged and dulwich is upgraded

**Comment by dtrifiro** [comment]
CI failures are related to linting, see #65

File: tests/test_git.py (hunk 1 above)
Comment (pmrowla) [reply tests/test_git.py:429]: looks like this should be fixed in the 0.20.38 which was just released?

Comment (dtrifiro) [reply tests/test_git.py:429]: Indeed, let me get rid of this, but before that, we need to get rid of the `file://` urls workaround for windows

Comment (dtrifiro) [reply tests/test_git.py:429]: https://github.com/iterative/scmrepo/pull/68

**Review by pmrowla — APPROVED** [review]

**Pull request merged**